Sources/SwiftFrameCore/Extensions/Collection+Extensions.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+extension Array where Element == URL {
+
+    func filter(pattern: String?) throws -> Self {
+        guard let pattern = pattern else {
+            return self
+        }
+        guard let regex = try? NSRegularExpression(pattern: pattern, options: .caseInsensitive) else {
+            throw NSError(description: "Invalid regular expression: \"\(pattern)\"")
```

Comment:
I wonder if we should check the validity of the regex when parsing the config file already?